File: IQ/Classes/clsBundleSystem.vb
Imports dataAccess

Public Class clsBundleSystem

    Property ID As Integer
    Property Bundle As clsBundle
    Property System As clsProduct
    Property rebate As Single

    Public Sub New(bundle As clsBundle, system As clsProduct, rebate As Single, Optional writecache As DataTable = Nothing)

        'price is some absolute overriding price on this bundle item (guaranteed by HP to be less then anyones buy price)
        'Rebate - is relative discount of their normal 'buy price' - rebate is subtracted - so use positive numbers

        If writecache Is Nothing Then
            Dim sql$
            sql$ = "INSERT INTO BundleSystem(fk_Bundle_id,fk_product_id_system,rebate) VALUES (" & bundle.ID & "," & bundle.ID & "," & system.ID & "," & rebate & ")"
            Me.ID = da.DBExecutesql(sql$)
            If system.Bundles Is Nothing Then system.Bundles = New Dictionary(Of Integer, clsBundle)
            system.Bundles.Add(bundle.ID, bundle)

        Else
            Me.ID = -1

            Dim row As System.Data.DataRow
            row = writecache.NewRow()

            row("fk_bundle_id") = bundle.ID
            row("fk_product_id_system") = system.ID
            row("rebate") = rebate
            writecache.Rows.Add(row)

        End If

        Me.Bundle = bundle
        Me.System = system
        Me.rebate = rebate

    End Sub

    Public Sub New(ID As Integer, bundle As clsBundle, system As clsProduct, rebate As Single)

        Me.ID = ID
        Me.Bundle = bundle
        Me.System = system
        Me.rebate = rebate

        If system.Bundles Is Nothing Then system.Bundles = New Dictionary(Of Integer, clsBundle)
        system.Bundles.Add(bundle.ID, bundle)


    End Sub


End Class

File: IQ/Classes/clsBundleSystem.cs
using dataAccess;


public class clsBundleSystem
{

    public int ID { get; set; }
    public clsBundle Bundle { get; set; }
    public clsProduct System { get; set; }
    public float rebate { get; set; }

    public clsBundleSystem(clsBundle bundle, clsProduct system, float rebate, DataTable writecache = null)
    {

        //price is some absolute overriding price on this bundle item (guaranteed by HP to be less then anyones buy price)
        //Rebate - is relative discount of their normal 'buy price' - rebate is subtracted - so use positive numbers

        if (writecache == null)
        {
            object sql = null;
            sql = "INSERT INTO BundleSystem(fk_Bundle_id,fk_product_id_system,rebate) VALUES (" + Bundle.ID + "," + Bundle.ID + "," + system.ID + "," + System.Convert.ToString(rebate) + ")";
            this.ID = System.Convert.ToInt32(da.DBExecutesql(sql));
            if (system.Bundles == null)
            {
                system.Bundles = new Dictionary<int, clsBundle>();
            }
            system.Bundles.Add(Bundle.ID, bundle);

        }
        else
        {
            this.ID = -1;

            System.Data.DataRow row = default(System.Data.DataRow);
            row = writecache.NewRow();

            row["fk_bundle_id"] = Bundle.ID;
            row["fk_product_id_system"] = system.ID;
            row["rebate"] = rebate;
            writecache.Rows.Add(row);

        }

        this.Bundle = bundle;
        this.System = system;
        this.rebate = rebate;

    }

    public clsBundleSystem(int ID, clsBundle bundle, clsProduct system, float rebate)
    {

        this.ID = ID;
        this.Bundle = bundle;
        this.System = system;
        this.rebate = rebate;

        if (system.Bundles == null)
        {
            system.Bundles = new Dictionary<int, clsBundle>();
        }
        system.Bundles.Add(Bundle.ID, bundle);


    }


}